sources/api/apiserver/src/server/mod.rs

```diff
@@ -947,15 +947,15 @@ impl ResponseError for error::Error {
             ConfigApplierWait { .. } => StatusCode::INTERNAL_SERVER_ERROR,
             ConfigApplierWrite { .. } => StatusCode::INTERNAL_SERVER_ERROR,
             SystemdNotify { .. } => StatusCode::INTERNAL_SERVER_ERROR,
-            SystemdNotifyStatus {} => StatusCode::INTERNAL_SERVER_ERROR,
+            SystemdNotifyStatus => StatusCode::INTERNAL_SERVER_ERROR,
             SetPermissions { .. } => StatusCode::INTERNAL_SERVER_ERROR,
             SetGroup { .. } => StatusCode::INTERNAL_SERVER_ERROR,
             SettingsToJson { .. } => StatusCode::INTERNAL_SERVER_ERROR,
             ReleaseData { .. } => StatusCode::INTERNAL_SERVER_ERROR,
             Shutdown { .. } => StatusCode::INTERNAL_SERVER_ERROR,
             Reboot { .. } => StatusCode::INTERNAL_SERVER_ERROR,
             UpdateDispatcher { .. } => StatusCode::INTERNAL_SERVER_ERROR,
-            UpdateError { .. } => StatusCode::INTERNAL_SERVER_ERROR,
+            UpdateError => StatusCode::INTERNAL_SERVER_ERROR,
             UpdateStatusParse { .. } => StatusCode::INTERNAL_SERVER_ERROR,
```

Comment:
Q - why did we not update other places to remove the unpack? Or is this to reflect a change to the original struct?

Reply:
The warning is only for unit structs:
```
warning: struct pattern is not needed for a unit variant
   --> api/apiserver/src/server/mod.rs:958:24
    |
958 |             UpdateError { .. } => StatusCode::INTERNAL_SERVER_ERROR,
    |                        ^^^^^^^ help: remove the struct pattern
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unneeded_struct_pattern
```